Deal with rate limit

statnmap opened the pull request:
You can retrieve the rate Limit that GitHub allows you with the code of this PR.  

I calculated the rate value necessary to download all ThinkR repositories, according to calculation of GitHub rate limit minutes.  
I saw that I needed at least a score of 1250 before starting to fetch repo. 
If you wait for the next hour when the rate is low, then the CI won't fail.

issue #35

statnmap committed:
Deal with rate limit

You can retrieve the rate Limit that GitHub allows you with the code of this PR.  

I calculated the rate value necessary to download all ThinkR repositories, according to calculation of GitHub rate limit minutes.  
I saw that I needed at least a score of 1250 before starting to fetch repo. 
If you wait for the next hour when the rate is low, then the CI won't fail.

jimhester commented:
Thanks for the PR!

This calculates the primary rate limit, but we are running into a secondary rate limit

statnmap commented:
I think you are using both graphql API and the GET API in the code. Both have two different ways of calculating a rate limit.
For the second one, you can directly use `gh::gh_rate_limit()` I guess.

statnmap committed:
Refine remaining rate at each step

On Github Actions, the maximum rate is 1000.  
Each step requiring points is calculated, and there may be waiting time if required.  
When using `rmarkdown:render()` the remaining points of each step is shown in the console for refinement for next time.

hadley commented:
I think we should handle this in gh, since it's a common need: https://github.com/r-lib/gh/issues/67.
